product-service/src/main/java/org/arpha/service/GenreServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package org.arpha.service;
+
+import com.querydsl.core.types.Predicate;
+import lombok.RequiredArgsConstructor;
+import org.arpha.dto.product.GenreDto;
+import org.arpha.exception.CreateGenreException;
+import org.arpha.exception.DeleteEntityException;
+import org.arpha.mapper.GenreMapper;
+import org.arpha.repository.GenreRepository;
+import org.arpha.utils.Boxed;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class GenreServiceImpl implements GenreService {
+
+    private final GenreRepository genreRepository;
+    private final GenreMapper genreMapper;
+    private final ProductService productService;
+
+    @Override
+    public GenreDto createGenre(GenreDto genreDto) {
+        return Boxed
+                .of(genreDto)
+                .mapToBoxed(genreMapper::toGenre)
+                .mapToBoxed(genreRepository::save)
+                .mapToBoxed(genreMapper::toGenreDto)
+                .orElseThrow(() -> new CreateGenreException("Error happened during genre creating"));
+    }
```

Comment:
What kind of error is it? 400/500. Should we provide more meaningful exception message?